File: sample/src/main/java/com/github/stephenvinouze/advancedrecyclerviewsample/activities/MainActivity.java
package com.github.stephenvinouze.advancedrecyclerviewsample.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;

import com.github.stephenvinouze.advancedrecyclerviewsample.R;
import com.github.stephenvinouze.advancedrecyclerviewsample.fragments.GestureRecyclerFragment;
import com.github.stephenvinouze.advancedrecyclerviewsample.fragments.MultipleChoiceRecyclerFragment;
import com.github.stephenvinouze.advancedrecyclerviewsample.fragments.PaginationRecyclerFragment;
import com.github.stephenvinouze.advancedrecyclerviewsample.fragments.SectionRecyclerFragment;
import com.github.stephenvinouze.advancedrecyclerviewsample.fragments.SingleChoiceRecyclerFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
        displaySingleChoiceRecyclerFragment();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.single_choice_action:
                displaySingleChoiceRecyclerFragment();
                break;

            case R.id.multiple_choice_action:
                displayMultipleChoiceRecyclerFragment();
                break;

            case R.id.section_action:
                displaySectionRecyclerFragment();
                break;

            case R.id.gesture_action:
                displayGestureRecyclerFragment();
                break;

            case R.id.pagination_action:
                displayPaginationRecyclerFragment();
                break;
        }

        return super.onOptionsItemSelected(item);
    }

    private void displaySingleChoiceRecyclerFragment() {
        setTitle(getString(R.string.single_choice_recycler_name));
        getSupportFragmentManager().beginTransaction().replace(R.id.main_container, new SingleChoiceRecyclerFragment()).commit();
    }

    private void displayMultipleChoiceRecyclerFragment() {
        setTitle(getString(R.string.multiple_choice_recycler_name));
        getSupportFragmentManager().beginTransaction().replace(R.id.main_container, new MultipleChoiceRecyclerFragment()).commit();
    }

    private void displaySectionRecyclerFragment() {
        setTitle(getString(R.string.section_recycler_name));
        getSupportFragmentManager().beginTransaction().replace(R.id.main_container, new SectionRecyclerFragment()).commit();
    }

    private void displayGestureRecyclerFragment() {
        setTitle(getString(R.string.gesture_recycler_name));
        getSupportFragmentManager().beginTransaction().replace(R.id.main_container, new GestureRecyclerFragment()).commit();
    }

    private void displayPaginationRecyclerFragment() {
        setTitle(getString(R.string.pagination_recycler_name));
        getSupportFragmentManager().beginTransaction().replace(R.id.main_container, new PaginationRecyclerFragment()).commit();
    }

}

File: sample/src/main/java/com/github/stephenvinouze/advancedrecyclerview/sample/activities/MainActivity.kt
package com.github.stephenvinouze.advancedrecyclerview.sample.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import com.github.stephenvinouze.advancedrecyclerview.sample.R
import com.github.stephenvinouze.advancedrecyclerview.sample.fragments.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        displaySingleChoiceRecyclerFragment()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.single_choice_action -> displaySingleChoiceRecyclerFragment()
            R.id.multiple_choice_action -> displayMultipleChoiceRecyclerFragment()
            R.id.section_action -> displaySectionRecyclerFragment()
            R.id.gesture_action -> displayGestureRecyclerFragment()
            R.id.gesture_section_action -> displayGestureSectionRecyclerFragment()
            R.id.pagination_action -> displayPaginationRecyclerFragment()
        }

        return super.onOptionsItemSelected(item)
    }

    private fun displaySingleChoiceRecyclerFragment() {
        title = getString(R.string.single_choice_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, SingleChoiceRecyclerFragment()).commit()
    }

    private fun displayMultipleChoiceRecyclerFragment() {
        title = getString(R.string.multiple_choice_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, MultipleChoiceRecyclerFragment()).commit()
    }

    private fun displaySectionRecyclerFragment() {
        title = getString(R.string.section_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, SectionRecyclerFragment()).commit()
    }

    private fun displayGestureRecyclerFragment() {
        title = getString(R.string.gesture_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, GestureRecyclerFragment()).commit()
    }

    private fun displayGestureSectionRecyclerFragment() {
        title = getString(R.string.gesture_sections_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, GestureSectionRecyclerFragment()).commit()
    }

    private fun displayPaginationRecyclerFragment() {
        title = getString(R.string.pagination_recycler_name)
        supportFragmentManager.beginTransaction().replace(R.id.main_container, PaginationRecyclerFragment()).commit()
    }

}
